src/traces/parcoords/defaults.js

```diff
@@ -0,0 +1,101 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Lib = require('../../lib');
+var attributes = require('./attributes');
+var hasColorscale = require('../../components/colorscale/has_colorscale');
+var colorscaleDefaults = require('../../components/colorscale/defaults');
+
+// todo unify with scatter/line_defaults.js (which needs to handle line width and dash as well)
+var handleLineDefaults = function lineDefaults(traceIn, traceOut, defaultColor, layout, coerce) {
+    var markerColor = (traceIn.marker || {}).color;
+
+    coerce('line.color', defaultColor);
+
+    if(hasColorscale(traceIn, 'line')) {
+        colorscaleDefaults(traceIn, traceOut, layout, coerce, {prefix: 'line.', cLetter: 'c'});
+    }
+    else {
+        var lineColorDflt = (Array.isArray(markerColor) ? false : markerColor) || defaultColor;
+        coerce('line.color', lineColorDflt);
+    }
+};
+
+
+function dimensionsDefaults(traceIn, traceOut) {
+    var dimensionsIn = traceIn.dimensions || [],
+        dimensionsOut = traceOut.dimensions = [];
+
+    var dimensionIn, dimensionOut;
+
+    function coerce(attr, dflt) {
+        return Lib.coerce(dimensionIn, dimensionOut, attributes.dimensions, attr, dflt);
+    }
+
+    for(var i = 0; i < dimensionsIn.length; i++) {
+        dimensionIn = dimensionsIn[i];
+        dimensionOut = {};
+
+        if(!Lib.isPlainObject(dimensionIn) || !Array.isArray(dimensionIn.values)) {
```

Comment:
Just summing up the current status of this feedback item - as code is right now -, which fell this way in part due to preexisting code regularity, but I'm glad to iterate, as there are some mismatches to the feedback, I just want to make sure work toward it is deemed valuable:

- The `dimensions[i].visible` attribute has been added.

- Each `dimension` has an `id` (defaults to the `label` if unsupplied) - on `Plotly.restyle`, dimensions of matching `id` are updated; dimensions with a new `id` are added. My pre-feedback assumption was that the user can add dimensions incrementally.

- Currently, there's no special handling for `values.length === 0` however the line count is determined as the minimum of `values.length` across dimensions, so in this case, no line gets rendered. I like the idea to handle `values.length === 0` as a special case, to implicitly mean that the dimension be removed. Also, the code already skips non-`isPlainObject` dimensions and dimensions whose `values` isn't an array which I think is what you meant; I'll reify both with a test case shortly.